### PR TITLE
[JENKINS-40806] tfs plugin PCT issues against 2.19.4 and 2.32.1

### PR DESCRIPTION
**Pull request by varyvol**
[JENKINS-40806](https://issues.jenkins-ci.org/browse/JENKINS-40806)

- Increase parent POM version.
- Increase baseline.
- Include necessary dependencies.
- Include "escape-by-default" property in Jelly files.
- Use MasterToSlaveCallable needed in new baseline version.

@yacaovsnc @DavidStaheli @mosabua @reviewbybees

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -3,20 +3,23 @@
...
-    <jenkins.version>1.580</jenkins.version>
+    <jenkins.version>1.625.1</jenkins.version>
+    <java.level>7</java.level>
     <maven-surefire-plugin.version>2.19.1</maven-surefire-plugin.version>
     <!--  overriding old version from parent to be consistent with the above in this project -->
     <maven-surefire-report-plugin.version>${maven-surefire-plugin.version}</maven-surefire-report-plugin.version>
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
     <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
+    <maven-javadoc-plugin.version>2.8</maven-javadoc-plugin.version>
+    <findbugs.failOnError>false</findbugs.failOnError>
```
Comment (varyvol) [inline pom.xml]: Raised [JENKINS-40807](https://issues.jenkins-ci.org/browse/JENKINS-40807) to take care of them.

**Comment by varyvol** [comment]
@olivierdagenais @DavidStaheli @mosabua did you have the opportunity to review this?

**Comment by varyvol** [comment]
@olivierdagenais @DavidStaheli @mosabua ping?

**Comment by varyvol** [comment]
@olivierdagenais @DavidStaheli @mosabua ping?

**Comment by olivierdagenais** [comment]
I regret to inform you that I have left this project. Try @yacaovsnc

**Comment by varyvol** [comment]
@olivierdagenais thanks for letting me know.

**Comment by yacaovsnc** [comment]
@varyvol Thanks for the pull request.  If I understand this pull request correctly, it drops Java 1.6 support and requires Java 7?  

There is a bug fix that we'd like to push out first without change the system requirement.  We will take a look at this PR soon.  

Thanks!

**Comment by varyvol** [comment]
@yacaovsnc it does because the baseline Jenkins versions requires Java 7.

No problem, thanks for the heads up!

**Comment by yacaovsnc** [comment]
```
Failed tests: 
  FunctionalTest.createLabel:224 expected:<NONE> but was:<INCOMPARABLE>
  FunctionalTest.oldPollingFallback:616 expected:<hudson.scm.PollingResult@434c179e> but was:<hudson.scm.PollingResult@46199519>
  FunctionalTest.upgradeEncodedPassword:666 expected:<NONE> but was:<INCOMPARABLE>
  FunctionalTest.useWebProxyServer:729 expected:<NONE> but was:<INCOMPARABLE>

Tests run: 12, Failures: 4, Errors: 0, Skipped: 1
```

There are functional test failures with this pull request.  We need more time to properly debug and merge this pull request.  Thanks for understanding.

**Comment by varyvol** [comment]
@yacaovsnc how are you executing the tests? I have merged the branch with the actual master and executed the tests with "mvn clean verify" and they seem to be working fine.

**Comment by yacaovsnc** [comment]
I followed this [doc](https://github.com/jenkinsci/tfs-plugin/blob/master/Testing.md).

**Comment by varyvol** [comment]
@yacaovsnc I see. Unfortunately I don't have a TFS to execute these functional tests against.

I am suspecting that maybe the two changes in Server class or the one in RemoteChangesetVersionCommand class should be SlaveToMasterCallable and not MasterToSlaveCallable. 

Would it be hard for you to change those and execute the tests?

**Comment by yacaovsnc** [comment]
@varyvol it's scheduled on my backlog to fix those changes.  Right now I am tied up with another priority and I will take a look at this soon.

**Comment by recena** [comment]
@yacaovsnc Any news?

**Comment by yacaovsnc** [comment]
@recena working on it today.

**Comment by varyvol** [comment]
@yacaovsnc thanks. Let me know if I can be of any help.

**Comment by yacaovsnc** [comment]
@varyvol The test failure doesn't look like an issue with the Callable changes, but rather a problem with upgrading the Jenkins version.  

In Jenkins 1.625, `AbstractLazyLoadRunMap.search(final int n, final Direction d)` seem are looking for exact matches; while in v1.58, it goes through a "slow path" to find the build from disk.  I verified the workspaces on disk, it seems in both versions the builds were fine.  Just in `AbstractProject.java:1309`, when we call "getLastBuild()", we are always getting `null`.  

How could we make search() call return with last build?

**Comment by yacaovsnc** [comment]
Will fix the tests failure in another pr.  Had to ignore one test as the project layout seems different and without the 'slow path' code path to find the build on disk, it's hard to change.

**Comment by varyvol** [comment]
@yacaovsnc thanks for investigating and working on this.

**Pull request merged**